### ISSUE TITLE
Multiplayer memory leak

### ISSUE THREAD

**Issue by roboderpy** [JavidPack/BossChecklist#4]
```
[15:26:46] [1/WARN] [tML]: Silently Caught Exception: 
System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.Collections.Generic.List`1.set_Capacity(Int32 value)
   at System.Collections.Generic.List`1.EnsureCapacity(Int32 min)
   at System.Collections.Generic.List`1.Add(T item)
   at BossChecklist.WorldAssist.PreUpdate() in BossChecklist\WorldAssist.cs:line 115
   at Terraria.ModLoader.WorldHooks.PreUpdate()
   at Terraria.WorldGen.UpdateWorld()
   at Terraria.Main.DoUpdate(GameTime gameTime)
   at Terraria.Main.Update(GameTime gameTime)
   at Terraria.Main.DedServ()
   at Terraria.Program.LaunchGame_()
   at Terraria.Program.LaunchGame(String[] args, Boolean monoArgs)
   at Terraria.WindowsLaunch.Main(String[] args)
```

**Comment by JavidPack** [comment]
Thanks, we'll look into it. Looking at the code I think you might be right.

**Comment by JavidPack** [comment]
Fixed in 089563fdfa4945004ff571510c8544c5799146b8
